### PR TITLE
refactor ext_utils

### PR DESCRIPTION
**Pull request by WaberZhuang**
Signed-off-by: 阡沉 <zhuangbowei.zbw@alibaba-inc.com>

### CODE REVIEW

File: src/extfs/extfs_utils.h [hunk 1]
```diff
@@ -222,117 +207,89 @@ static int rmdir_proc(
...
         case EXT2_ET_BAD_MAGIC:
-            decode = "EXT2_ET_BAD_MAGIC";
-            break;
+            err_msg = "EXT2_ET_BAD_MAGIC";
+            return -EIO;
+
         case EXT2_ET_DIR_NO_SPACE:
-            decode = "EXT2_ET_DIR_NO_SPACE";
-            break;
+            err_msg = "EXT2_ET_BAD_MAGIC";
```
Comment (yuchen0cc) [inline src/extfs/extfs_utils.h:271]: EXT2_ET_DIR_NO_SPACE

File: src/extfs/extfs_utils.h [hunk 2]
```diff
@@ -222,117 +207,89 @@ static int rmdir_proc(
...
+            return -err; /* fallthrough */
```
Comment (yuchen0cc) [inline src/extfs/extfs_utils.h:272]: ENOSPC

**Review by yuchen0cc — COMMENTED** [review]
reviews

File: src/extfs/extfs_utils.h [hunk 3]
```diff
@@ -222,117 +207,89 @@ static int rmdir_proc(
...
-        LOG_ERROR("ext2fs unclassified error: at `:`, ecode `:`", file, line, err, decode.c_str());
+            err_msg = "EXT2_ET_UNEXPECTED_BLOCK_SIZE";
+            return -EIO;
 
-    if (fs) {
-        ext2fs_mark_super_dirty(fs);
-        ext2fs_flush(fs);
+        default:
+            err_msg = "to be decode";
+            return -err;
```
Comment (yuchen0cc) [inline src/extfs/extfs_utils.h:284]: default EIO

File: src/extfs/extfs_utils.h [hunk 4]
```diff
@@ -141,72 +135,63 @@ static int unlink_file_by_name(ext2_filsys fs, const char *path) {
...
-            inode.i_dtime = time(0);
-            break;
-        default:
-            inode.i_links_count--;
-    }
+    if (ret) return parse_extfs_error(fs, ino, ret);
+
+    if (inode.i_links_count == 0) return 0;
+    inode.i_links_count--;
+    if (inode.i_links_count == 0) inode.i_dtime = time(0);
```
[the remark this answers is not in the document]
Comment (WaberZhuang) [reply src/extfs/extfs_utils.h:154]: ==1 的话会走第二个 ==0

[the remark this answers is not in the document]
Comment (yuchen0cc) [reply src/extfs/extfs_utils.h:154]: OK

**Comment by yuchen0cc** [comment]
LGTM

**Pull request merged**